bot/exts/help_channels.py

```diff
@@ -542,18 +546,25 @@ async def move_to_available(self) -> None:
 
     async def move_to_dormant(self, channel: discord.TextChannel, caller: str) -> None:
         """
-        Make the `channel` dormant.
+        Make the `channel` dormant and remove the help cooldown role if it was the claimant's only channel.
 
         A caller argument is provided for metrics.
         """
         log.info(f"Moving #{channel} ({channel.id}) to the Dormant category.")
 
+        guild = self.bot.get_guild(constants.Guild.id)
+        claimant = guild.get_member(await self.help_channel_claimants.get(channel.id))
+
         await self.help_channel_claimants.delete(channel.id)
         await self.move_to_bottom_position(
             channel=channel,
             category_id=constants.Categories.help_dormant,
         )
 
+        # Remove the cooldown role if the claimant has no other channels left
+        if claimant.id not in {user_id for _, user_id in await self.help_channel_claimants.items()}:
```

Comment:
It may be more efficient to use `any()` and compare each ID while looping than to create a set, since the former would be able to exit early when a match is found.